concat_yaml.py

```diff
@@ -0,0 +1,321 @@
+
+warnings.simplefilter('ignore', yaml.error.UnsafeLoaderWarning)
+
+# CONSISTENCY CHECKS
+
+
+def allign_cuda_versions(joint_packages, module_packages_file, version):
+    '''
+    Take to prefix provided by spack-config and replace the one
+    taken from sysconfig/templates
+    '''
+
+    print('Allign cuda versions')
+
+    module_packages = load_from_yaml(module_packages_file)['packages']
+
+    cuda_joint = joint_packages['packages']['cuda']
+
+    spec_joint = cuda_joint['externals'][0]['spec']
+
+    if version not in spec_joint:
+        raise ValueError(
+            f'Cuda version {version} not provided by yaml from templates')
+
+    specs_module = []
+    prefix_module = []
+    for i in range(len(module_packages['cuda']['externals'])):
+        specs_module.append(module_packages['cuda']['externals'][i]['spec'])
+        prefix_module.append(module_packages['cuda']['externals'][i]['prefix'])
+
+    i = 0
+    found_cuda_version = False
+    for spec in specs_module:
+        if version in spec:
+            prefix = prefix_module[i]
+            found_cuda_version = True
+            break
+        i += 1
+
+    if not found_cuda_version:
+        raise ValueError(
+            f'Cuda version {version} not provided by spack-config module')
+
+    joint_packages['packages']['cuda']['externals'][0]['prefix'] = prefix
+
+    return joint_packages
+
+
+def rename_cray_mpich_to_mpich(packages):
+    '''
+    Rename cray-mpich from spack-config module
+    to mpich to be compatible with spack-c2sm
+    '''
+
+    print('Rename cray-mpich to mpich')
+    cray_mpich = packages['packages']['cray-mpich']
+
+    spec = cray_mpich['externals'][0]['spec']
+    spec = spec.replace('cray-', '')
+
+    cray_mpich['externals'][0]['spec'] = spec
+
+    packages['packages']['mpich'] = cray_mpich
+
+    packages['packages']['mpich']['buildable'] = False
+
+    packages['packages'].pop('cray-mpich')
+
+    return packages
+
+
+def allow_xml_to_be_built(packages):
+    print('Allow building of xml')
+    packages['packages']['libxml2']['buildable'] = True
+    return packages
+
+
+# SPACK COMMANDS
+
+
+def spack_external_find(machine, packages_file):
+    '''
+    run spack external find and write
+    packages.yaml to current workingdir
+    '''
+
+    print(f'Find externals on {machine}')
+
+    os.environ["SPACK_USER_CONFIG_PATH"] = os.getcwd()
+
+    if os.path.exists(packages_file): os.remove(packages_file)
+
+    command = [
+        "./config.py",
+        "-i",
+        ".",
+        "-u",
+        "OFF",
+        "-m",
+        machine,
+        "--no_yaml_copy",
+    ]
+    subprocess.run(command, check=True)
+    command = [
+        'bash', '-c', "source spack/share/spack/setup-env.sh && \
+               spack external find --not-buildable --scope=user"
+    ]
+    subprocess.run(command, check=True)
+
+    os.environ.pop("SPACK_USER_CONFIG_PATH")
+
+
+# MERGE OF INDIVIDUAL YAML-FILES
+
+
+def remove_duplicate_compilers(c2sm, cscs, keys):
```

Comment:
The name of this function is misleading, it doesn't remove anything. It returns something.
How does `cscs` suddenly come into play?

AFAIK this achieves the same, right?
c2sm_specs = { item[key_1][key_2] for item in c2sm }
return c2sm + [ item for item in cscs if item[key_1][key_2] not in c2sm_specs]

May I propose to replace the function with
```
def unique_compilers(primary, secondary, key_1, key_2):
    primary_specs = { item[key_1][key_2] for item in primary }
    return primary + [ item for item in secondary if item[key_1][key_2] not in primary_specs]
```
Feel free to give it a better name.

Reply:
disambiguate_compilers or disambiguate_compilers_with_precedence, maybe?